pypesto/petab/importer.py

```diff
@@ -415,6 +436,37 @@ def create_objective(
             par_ids = [x for x in par_ids if x not in inner_parameter_ids]
 
             # FIXME: currently not supported with hierarchical
+            if 'guess_steadystate' in kwargs and kwargs['guess_steadystate']:
+                warnings.warn(
+                    "`guess_steadystate` not supported with hierarchical optimization. Disabling `guess_steadystate`."
+                )
+            kwargs['guess_steadystate'] = False
+
+        if self._ordinal:
+            inner_solver_options = kwargs.pop('inner_solver_options', None)
+            inner_solver_options = (
+                inner_solver_options
+                if inner_solver_options is not None
+                else self._inner_solver_options
+            )
+            inner_problem_method = inner_solver_options.get('method', None)
+
+            inner_problem = OptimalScalingProblem.from_petab_amici(
+                self.petab_problem, model, edatas, inner_problem_method
+            )
+            inner_solver = OptimalScalingInnerSolver(
```

Comment:
I don't understand why construction of inner problem/solver is handled differently from the standard hierarchical setting. Wouldn't it make more sense to implement inner solver option handling more generically?

Reply:
`OptimalScalingProblem` and `OptimalScalingParameter` are implemented as child classes of the generic `InnerProblem` and `InnerParameter` that are used for relative data (scaling, offset, noise). 

On the other hand, `OptimalScalingSolver` and `OptimalScalingCalculator` are not since they're completely different from the calculator and solvers used for relative data. 

So because some of the classes are completely separate, I implemented the construction of everything completely separate too. I agree it's not nice and would like it to be nicer. Especially it won't be nice when I PR the spline method for nonlinear monotone data too. 

So I'm open to suggestions now, and then, on how to make it nicer.

Reply:
okay should then be fine to refactor later on.